packages/flutter_tools/lib/src/commands/run_mojo.dart

```diff
@@ -87,20 +88,28 @@ class RunMojoCommand extends FlutterCommand {
     final String command = useDevtools ? _getDevtoolsPath() : _getMojoShellPath();
     args.add(command);
 
+    BuildConfiguration config = _getCurrentHostConfig();
+
     if (argResults['android']) {
       args.add('--android');
-      final String cloudStorageBaseUrl = ArtifactStore.getCloudStorageBaseUrl('shell', 'android-arm');
       final String appPath = _makePathAbsolute(bundlePath);
       final String appName = path.basename(appPath);
       final String appDir = path.dirname(appPath);
       args.add('http://app/$appName');
       args.add('--map-origin=http://app/=$appDir');
-      args.add('--map-origin=http://flutter/=$cloudStorageBaseUrl');
-      args.add('--url-mappings=mojo:flutter=http://flutter/flutter.mojo');
+      if (config == null || config.type == BuildType.prebuilt) {
+        final String cloudStorageBaseUrl = ArtifactStore.getCloudStorageBaseUrl('shell', 'android-arm');
+        args.add('--map-origin=http://flutter/=$cloudStorageBaseUrl');
+        args.add('--url-mappings=mojo:flutter=http://flutter/flutter.mojo');
+      } else {
+        String localPath = path.join(config.buildDir, 'flutter.mojo');
+        File flutterMojo = new File(localPath);
+        flutterMojo.copySync(path.join(appDir, 'flutter.mojo'));
```

Comment:
I don't think we want to copy this file into the developer's directory.  Can we add another URL mapping to the devtools web server to serve this file out of its current location?